### ISSUE TITLE
[Magisk] SafetyNet pass

### ISSUE THREAD

**Issue by cawilliamson** [phhusson/treble_experimentations#116]
Ok so I've read a lot of conflicting points of view on this but from my research around the Interwebs it seems that a few things are true:

1.) This GSI doesn't currently pass SafetyNet due to "ctsProfile: false" meaning a CTS profile mismatch.

2.) Magisk will absolutely never pass out-of-the-box on a GSI without any tweaking at all unless one is made specifically for a single device *or* has a release fingerprint ID of a working device.

3.) Play Store will always show "Uncertified" while SafetyNet is not passing.

4.) **MOST IMPORTANTLY** Some folks *DO* have MagiskHide working on GSI / Treble images by using the "MagiskHide Props Config", running "su", then "propts" and selecting their device to install its release key to build.prop

So in short - the problem I'm seeing is that this GSI *can not* pass CTS profile but other GSI images can ( such as https://forum.xda-developers.com/project-treble/trebleenabled-device-development/8-1-0-pixel-experience-t3796011 ) as multiple comments and screenshots provide proof of.

This also appears to affect all Phh based builds so I can only assume the issue is in one of the base repos rather than distro-specific.

**Comment by cawilliamson** [comment]
I've cracked this one!

There's no super-fancy detection going on here - on my OnePlus 6 I managed to pass SafetyNet by doing the following:

1.) Install Magisk 16.4 (nothing special here - no fancy builds needed.)
2.) Install the "MagiskHide Props Config" module
3.) Reboot
4.) Run a terminal app and type:

```
su
props
```

5.) Set the fingerprint with the following keys (pressing enter after each)
1 -> f -> 7 -> 3 -> y -> n -> b -> b -> b

5.) Enable Improved hiding:

2 -> y -> n

6.) Add/edit custom props:

4 -> n -> ro.build.ab_update -> true -> y -> n
4 -> n -> ro.product.brand -> OnePlus -> y -> n
4 -> n -> ro.product.manufacturer -> OnePlus -> y -> n
4 -> n -> ro.product.model -> ONEPLUS A5010 -> y -> n -> b

7.) Script settings:

6 -> 1 -> y -> n -> b

8.) Reboot:

b -> y

After this SafetyNet passed perfectly for me so it looks like the SafetyNet issue is just entirely based on what is in the build.prop for any given device. Fix those and it passes straight away. 👍

**Comment by P1N2O** [comment]
This investigation of yours helped a ton mate!
Thanks!

**Comment by vineet1122** [comment]
@cawilliamson 7th step didnt worked for me?? it says input a valid command..i'm totally new to this kindly help!

**Comment by cawilliamson** [comment]
@vineet1122 Copy/pasting wont work dude.

As for step 5 - skip it - if you read next to it - I wrote this step isn't needed (just left there for the purposes of the bug report.)

The key thing is step 7 is to change from late_start service to post-fs-data. The numbers may have changed but with that info you should be able to adapt them for your needs.

Hope that helps!

**Comment by KRi570F** [comment]
the "MagiskHide Props Config" module has been updated,
any as now the steps are incorrect, can you please update those steps?

**Comment by gabrieloliveirabrito** [comment]
> the "MagiskHide Props Config" module has been updated,
> any as now the steps are incorrect, can you please update those steps?

On newer versions, i skipped the step 7, and on 6, set **boot-stage** to **post**

**Comment by deilonsn** [comment]
This fix is not working anymore on Magisk 22.1 :(
